test-browser/tests/solidityUnittests.js

```diff
@@ -33,3 +117,171 @@ function runTests (browser) {
+          return Assert.equal(foo.get(), 200, "value is not 200");
+        }
+
+        function shouldFailForWrongValue200() public returns (bool) {
+          foo.set(300);
+          return Assert.equal(foo.get(), 200, "value is not 200");
+        }
+      }
+        `
+    },
+    'browser/ks2a.sol': {
+      content: `
+      pragma solidity >=0.4.22 <0.6.0;
+      contract Kickstarter {
+          enum State { Started, Completed }
+      
+          struct Project {
+              address owner;
+              string name;
+              uint goal;
+              uint fundsAvailable; // added
+              uint amountContributed; // added
+              State state;
+              mapping(address => uint) funders; // added
+          }
+      
+          Project[] public projects;
+      
+          constructor() public {
+          }
+      
+          function createProject(string memory name, uint goal) public {
+              projects.length++; // new line
+              Project storage project = projects[projects.length - 1];
+              project.name = name;
+              project.goal = goal;
+              project.owner = msg.sender;
+              project.state = State.Started;
+          }
+          
+          function fundProject(uint projectId) payable public {
+          Project storage project = projects[projectId];
+              // require project exists
+              // PLEASE CHECK / or erase
+              // not this: require(projects[projectId].exists, "the project must exist to be funded");
+      
+              // require for... underflow/overflow protection
+              project.funders[msg.sender] += msg.value;
+              project.amountContributed += msg.value;
+              project.fundsAvailable += msg.value;
+      
+              if (project.amountContributed >= project.goal) {
+                  project.state = State.Completed;
+              }
+          }
+          
+          // this function is here because we can't use web3 when using the VM
+          function getContractBalance() public view returns(uint balance) {
+              return address(this).balance;
+          }
+            
+      }
+        `
+    },
+    'browser/ks2b_test.sol': {
+      content: `
+      pragma solidity >=0.4.22 <0.6.0;
+      pragma experimental ABIEncoderV2;
+
+      import "remix_tests.sol"; // this import is automatically injected by Remix.
+      import "remix_accounts.sol";
+      import "./ks2a.sol";
+
+      contract kickstarterTest {
+          enum State { Started, Completed }
+
+          Kickstarter kickstarter;
+          /// #sender: account-0
+          function beforeAll () public {
+            kickstarter = new Kickstarter();
+            kickstarter.createProject("ProjectA", 123000);
+            kickstarter.createProject("ProjectB", 100);
+          }
+      
+          /// #sender: account-0
+          /// #value: 10000000
+          function checkProjectExists () public payable {
```

Comment:
Its good to check expected value of sender and value inside test.  Use:

```
Assert.equal(msg.sender, TestsAccounts.getAccount(0), "wrong sender");
Assert.equal(msg.value, 10000000, "wrong value");
```

Also better to use any other account except account-0 as that is a default one.

Further, its good if we can have a test failing because of `wrong sender` or `wrong value`

Reply:
I'm having troubles testing multiple failing cases in a test because the test exits and skips to next test after encountering a single failing case.

Reply:
This is tracked further here: https://github.com/ethereum/remix-ide/issues/2745

Reply:
I created separate tests in the contract to handle the failing cases.